src/libmusig2.h

```diff
@@ -7,212 +7,141 @@
- *  Parameters  : IN        : ctx, a secp256k1_context object.
- *                          : X_LIST, the list of public keys.
- *                          : a_LIST, the list of exponents.
- *                          : N, number of signers.
- *              : IN/OUT    : X_, the aggregated public key.
- *                          : xonly_X_, the x_only aggregated public key.
- *                          : ser_xonly_X_, the serialized x_only aggregated public key.
- *                          : parity_X_, parity of xo_X_.
- * Returns      : 1/0
+typedef struct {
+    musig2_context *mc;
+    secp256k1_keypair **commlist;
+    secp256k1_keypair keypair;
+    secp256k1_pubkey agg_R_list[V];
+}musig2_context_sig;
+
+/** Struct      : musig2_param
+ *  Purpose     : Stores the parameters to generate a partial signature.
+ *  Parameters  : a: The exponent `a` of signer.
+ *              : b: nonce, b = H_non(X, (R_1, ..., R_V), msg).
+ *              : c: challenge, c = H_sig(X, R, msg).
+ *              : b_list: The list of `b` values, b_LIST = { b^(j-1) }.
+ *              : msg: The message to be signed.
+ *              : sx_X: Serialized x_only aggregated public key.
+ *              : sx_R: Serialized x_only aggregated R.
+ *              : par_R: Parity of R.
+ *              : par_X: Parity of X.
+ *              : msg_len: The length of message.
  * */
-int MuSig2_KeyAgg(const secp256k1_context *ctx,
-                  secp256k1_pubkey *X_LIST, secp256k1_pubkey *X_,
-                  secp256k1_xonly_pubkey *xonly_X_,
-                  unsigned char *ser_xonly_X_, unsigned char **a_LIST,
-                  int *parity_X_, int N) ;
-/*----------------------------------------------------------------------------------------------------------*/
-
-
-/** Function    : MuSig2_BatchCommitment
- *  Purpose     : Generates random nonce values and stores in r_LIST. Computes corresponding commitments and stores
- *                in R_LIST. Total number of nonce-commitment is NR_MSGS * V * N.
- *  Parameters  : IN        : ctx, a secp256k1_context object.
- *                          : NR_MSGS, the number of messages.
- *                          : V, the number of nonce values.
- *              : IN/OUT    : R_LIST, the list of public commitments.
- *                          : r_LIST, the list of secret nonce values.
- *                          : cnt, the last index of commitment list to assign next entry.
+typedef struct{
+    unsigned char a[SCALAR_BYTES];
+    unsigned char b[SCALAR_BYTES];
+    unsigned char c[SCALAR_BYTES];
+    unsigned char *b_LIST[V];
+    unsigned char *msg;
+    unsigned char sx_X[XONLY_BYTES];
+    unsigned char sx_R[XONLY_BYTES];
+    int par_R;
+    int par_X;
+    int msg_len;
+}musig2_param;
+
+/** Struct      : musig2_partial_signatures
+ *  Purpose     : Stores the parameters to aggregate a partial signatures.
+ *  Parameters  : sig: The partial signature of a signer.
+ *              : R_: The nonce of a signer.
```

Comment:
Again, why do we use underscore? And this structure is of a single signature (in singular), instead of signatures.